### PR TITLE
Feature/mds 2263

### PR DESCRIPTION
**Pull request by thegentlemanphysicist**
The incidents dashboard is almost complete, but there are a few things to be fixed

- frontend sort seems to be breaking

- there may be some issues loading the correct page from a copy pasted url

- edititing the incidents from the dashboard modal does not work yet (form values do not populate properly).  I may just hide the edit button

### CODE REVIEW

File: frontend/src/components/Forms/incidents/IncidentSearchForm.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                  data={this.props.incidentStatusCodeOptions}
+                />
+              </Col>
+              <Col md={12} xs={24}>
+                <Field
+                  id="year"
+                  name="year"
+                  placeholder="Select year of incident"
+                  component={renderConfig.YEAR}
+                  // validate={[dateNotInFuture]} //TODO IMPLEMENT DATE NOT IN FUTURE VALIDATION
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/Forms/incidents/IncidentSearchForm.js]: There was no year not in future, but there is now

File: frontend/src/components/dashboard/customHomePage/VarianceTable.js [hunk 2]
```diff
@@ -37,8 +37,6 @@ const defaultProps = {
 };
 export const VarianceTable = (props) => (
   <div className="tab__content">
-    <h4>Variances</h4>
-    <br />
     <MineVarianceTable
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/customHomePage/VarianceTable.js]: yes, I was matching it up to the figma docs.

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 3]
```diff
@@ -0,0 +1,378 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: ha ha ha sorry man

[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: fixed

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+const defaultProps = {
+  updateMineIncident: () => {},
+  followupActions: [],
+};
+
+export const joinOrRemove = (param, key) => {
+  if (isEmpty(param)) {
+    return {};
+  }
+  return typeof param === "string" ? { [key]: param } : { [key]: param.join(",") };
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: When I tried implementing it with variances it was not playing well with clearing the multi-select dropdown empty fields.  I agree, the new pattern is way way better, however I fixing it with dropdowns has been frustrating.

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
+  followupActions: PropTypes.arrayOf(CustomPropTypes.incidentFollowupType),
+  followupActionsOptions: CustomPropTypes.options.isRequired,
+  incidentDeterminationOptions: CustomPropTypes.options.isRequired,
+  incidentStatusCodeOptions: CustomPropTypes.options.isRequired,
+  doSubparagraphOptions: CustomPropTypes.options.isRequired,
+};
+
+const defaultProps = {
+  updateMineIncident: () => {},
```
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: fixed

File: frontend/src/components/Forms/incidents/IncidentSearchForm.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                  data={this.props.incidentStatusCodeOptions}
+                />
+              </Col>
+              <Col md={12} xs={24}>
+                <Field
+                  id="year"
+                  name="year"
+                  placeholder="Select year of incident"
+                  component={renderConfig.YEAR}
+                  validate={[yearNotInFuture]} // TODO IMPLEMENT DATE NOT IN FUTURE VALIDATION
```
Comment (NWCalvank) [inline frontend/src/components/Forms/incidents/IncidentSearchForm.js]: Is this TODO done now?

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+
+export const joinOrRemove = (param, key) => {
+  if (isEmpty(param)) {
+    return {};
+  }
+  return typeof param === "string" ? { [key]: param } : { [key]: param.join(",") };
+};
+export const removeEmptyStings = (param, key) => (isEmpty(param) ? {} : { [key]: param });
+
+// TODO: Implement the NoW dashboard patern for cleaning props
```
Comment (NWCalvank) [inline frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: *pattern

File: python-backend/app/api/incidents/resources/incidents_resource.py [hunk 8]
```diff
@@ -128,6 +132,10 @@ def _apply_filters_and_pagination(self, args):
 
         # Apply sorting
         if args['sort_field'] and args['sort_dir']:
+            logging.warning(f'the sort dir is: { args["sort_dir"]}')
+            logging.warning(f'the sort field is: { args["sort_field"]}')
+            logging.warning(f'the sort field is: { sort_field[args["sort_field"]]}')
+            logging.warning(f'the sort model is: { sort_models[args["sort_field"]]}')
```
Comment (NWCalvank) [inline python-backend/app/api/incidents/resources/incidents_resource.py]: Please remove all the logging in this file (and the logging import) 🙂

Comment (thegentlemanphysicist) [reply python-backend/app/api/incidents/resources/incidents_resource.py]: removed

File: frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js (hunk 7 above)
Comment (thegentlemanphysicist) [reply frontend/src/components/dashboard/incidentsHomePage/IncidentsHomePage.js]: done

File: frontend/src/components/Forms/incidents/IncidentSearchForm.js (hunk 6 above)
Comment (thegentlemanphysicist) [reply frontend/src/components/Forms/incidents/IncidentSearchForm.js]: removed

**Pull request merged**